storage/test/sparse_merkle/internal_node_test.cpp

```diff
@@ -16,6 +16,19 @@
 
 using namespace concord::storage::sparse_merkle;
 
+// Return a new hash with the given bit flipped for the given hash.
+//
+// Flip the most significant bit. So bit 0 to be flipped would be the first bit,
+// bit 1, the second bit, etc...
+Hash flipMSBit(size_t bit, const Hash& hash) {
+  Assert(bit < 8);
+  // Flip the first bit so that key1_hash becomes a sibling of key2_hash
+  std::array<uint8_t, Hash::SIZE_IN_BYTES> flipped;
+  std::copy(hash.data(), hash.data() + hash.size(), flipped.begin());
+  flipped[0] ^= (1 << (7 - bit));
+  return Hash(flipped);
+}
+
```

Comment:
[Minor]: Maybe this can be named differently as it is not always the MS bit. Something like`flipByte0Bit()`?